src/Config/DatabasePrimitives/DatabaseObject.cs

```diff
@@ -316,12 +387,27 @@ public override int GetHashCode()
 [DebuggerDisplay("ReferencingDbTable = {ReferencingDbTable.FullName}, ReferencedDbTable = {ReferencedDbTable.FullName}")]
 public class RelationShipPair
 {
+    /// <summary>
+    /// Relationship name specified in the runtime config.
+    /// </summary>
+    public string RelationshipName { get; set; } = string.Empty;
+
     public RelationShipPair() { }
 
     public RelationShipPair(
```

Comment:
Yes, there are still references to all constructors (during validations andoptimistic foreignkeydefinition creation scenarios). It will be a gradual adoption through the codebase to use the constructor that allows the dev to provide the relationship name. However, at this time, not all locations have a reference to the relationship name when `RelationshipPair` is created :

- `RelationShipPair(
        string relationshipName,
        DatabaseTable referencingDbObject,
        DatabaseTable referencedDbObject)`
-   `public RelationShipPair() { }`
- `public RelationShipPair(
DatabaseTable referencingDbObject,
DatabaseTable referencedDbObject)`